federation/develop/develop.sh

```diff
@@ -43,10 +43,15 @@ readonly TMP_DIR="$(mktemp -d)"
 readonly FEDERATION_OUTPUT_ROOT="${LOCAL_OUTPUT_ROOT}/federation"
 readonly VERSIONS_FILE="${FEDERATION_OUTPUT_ROOT}/versions"
 
-detect-project
-readonly KUBE_PROJECT="${KUBE_PROJECT:-${PROJECT:-}}"
+if [[ -n "${LOCAL_BUILD:-}" ]]; then
```

Comment:
Can we put this behind the `$KUBERNETES_PROVIDER` environment variable like the way we do here - https://github.com/kubernetes/kubernetes/blob/master/cluster/kubectl.sh?

This is not just about local vs cloud. This affects people who are using AWS or other non-GCP cloud providers. `detect-project` is specific to GCP.